### PR TITLE
feat: Hover with Markdown infomation

### LINKED ISSUE #45: feat: Hover could contain more infomation
Hello, I am a developer from [tensorchord](https://github.com/tensorchord). We use starlark-lsp as a denepndency to build our [LSP Server](https://github.com/tensorchord/envd-lsp). 
Much thanks for this project and its contributors!

I have a feature recommendation for starlark-lsp. If you think it is acceptable, I would be happy to help on it. If it could be merged, I could upgrade our project to introduce it either.

When we use `Python` extension, it would display `Parameters` and `Returns` for functions.
image

Hovever, `starlark-lsp` could only show first line(document) for `Hover`, it would be better for user if `Hover` could output arguments types.

Related issue: https://github.com/tensorchord/envd-lsp/issues/35

### PR DESCRIPTION
**Pull request by cutecutecat**
close #45 

Signed-off-by: cutecutecat <starkind1997@gmail.com>

Before:
image

After:
image

### CODE REVIEW

**Comment by cutecutecat** [comment]
@nicksieger PTAL

File: pkg/analysis/hover_test.go [hunk 1]
```diff
@@ -61,6 +61,27 @@ foo(hello)
...
+    name: name of the bar
+  Returns:
+    name of the foo
+  """
+  pass
+
+foo(hello)
+`)
+	result := f.a.Hover(f.ctx, doc, protocol.Position{Line: 12, Character: 2})
+	assertHoverResult(t, doc, "foo", "foos a bar\n# Parameters\nname: name of the bar\n# Returns\nname of the foo", result)
```
Comment (nicksieger) [inline pkg/analysis/hover_test.go:82]:
```suggestion
	assertHoverResult(t, doc, "foo", "foos a bar\n## Parameters\nname: name of the bar\n## Returns\nname of the foo", result)
```

File: pkg/query/signature.go [hunk 2]
```diff
@@ -99,10 +99,22 @@ func (s Signature) Label() string {
...
+	returns := s.Docs.Returns()
+	for _, arg := range s.Docs.Args() {
+		argsFormatted += fmt.Sprintf("\n%s: %s", arg.Name, arg.Desc)
+	}
+	detail := s.Docs.Description
+	if len(argsFormatted) > 0 {
+		detail += fmt.Sprintf("\n# Parameters%s", argsFormatted)
+	}
+	if len(returns) > 0 {
+		detail += fmt.Sprintf("\n# Returns\n%s", returns)
```
Comment (nicksieger) [inline pkg/query/signature.go:112]:
```suggestion
		detail += fmt.Sprintf("\n## Parameters%s", argsFormatted)
	}
	if len(returns) > 0 {
		detail += fmt.Sprintf("\n## Returns\n%s", returns)
```

**Review by nicksieger — APPROVED** [review]
Looks great! Yeah, I think level one `#` is a little big, let's use two?

**Review by nicksieger — APPROVED** [review]

**Comment by nicksieger** [comment]
@cutecutecat can you force-push one more time? We enabled builds for PRs from forked repos. Just need to get the ci check to run and I'll merge.

**Comment by cutecutecat** [comment]
> @cutecutecat can you force-push one more time? We enabled builds for PRs from forked repos. Just need to get the ci check to run and I'll merge.

It's done. @nicksieger

**Pull request merged**